Tests/SonarScanner.MSBuild.PreProcessor.Test/JreCaching/JreCacheTests.cs

```
@@ -538,6 +553,295 @@ public async Task Checksum_DownloadFile_FileOpenFails()
+            x => x.Should().StartWith(@"The extraction of the downloaded Java runtime environment failed with error 'The java executable in the extracted Java runtime environment was expected to be at 'C:\Users\user\.sonar\cache\sha256"));
+    }
+
+    [TestMethod]
+    public async Task UnpackerFactory_ReturnsNull()
+    {
+        var home = @"C:\Users\user\.sonar";
+        var cache = Path.Combine(home, "cache");
+        var sha = Path.Combine(cache, "sha256");
+        directoryWrapper.Exists(cache).Returns(true);
+        directoryWrapper.Exists(sha).Returns(true);
+        unpackerFactory.CreateForArchive(directoryWrapper, fileWrapper, TestArchiveName).ReturnsNull();
+
+        var sut = CreateSutWithSubstitutes();
+        var result = await sut.DownloadJreAsync(home, new(TestArchiveName, "sha256", "javaPath"), () => Task.FromResult<Stream>(new MemoryStream()));
+        result.Should().BeOfType<JreCacheFailure>().Which.Message.Should().Be("The archive format of the JRE archive `filename.tar.gz` is not supported.");
+        fileWrapper.DidNotReceiveWithAnyArgs().Exists(null);
+        fileWrapper.DidNotReceiveWithAnyArgs().Create(null);
+        fileWrapper.DidNotReceiveWithAnyArgs().Open(null);
+        checksum.DidNotReceiveWithAnyArgs().ComputeHash(null);
+        unpackerFactory.Received(1).CreateForArchive(directoryWrapper, fileWrapper, TestArchiveName);
+        testLogger.DebugMessages.Should().BeEmpty();
+    }
+
+    [TestMethod]
+    public async Task UnpackerFactory_Throws()
+    {
+        var home = @"C:\Users\user\.sonar";
+        var cache = Path.Combine(home, "cache");
+        var sha = Path.Combine(cache, "sha256");
+        directoryWrapper.Exists(cache).Returns(true);
+        directoryWrapper.Exists(sha).Returns(true);
+        unpackerFactory.CreateForArchive(directoryWrapper, fileWrapper, TestArchiveName).ReturnsNull();
```

Shouldn't this throw instead of return null?

---

That is what it did before, but you requested to change it: https://github.com/SonarSource/sonar-scanner-msbuild/pull/2018#discussion_r1668511430

---

Oh, yeah, sry. The test name is not up to date and it confused me.

---

I corrected the test name.